extensions/json/client/src/jsonMain.ts

```diff
@@ -244,16 +246,16 @@ function getSettings(): Settings {
 	};
 
 	// merge global and folder settings. Qualify all file matches with the folder path.
-	let globalSettings = workspace.getConfiguration('json', null).get<JSONSchemaSettings[]>('schemas');
-	if (Array.isArray(globalSettings)) {
-		collectSchemaSettings(globalSettings, workspace.rootPath);
+	let globalSettings = workspace.getConfiguration('json').inspect<JSONSchemaSettings[]>('schemas');
```

Comment:
This part required changing the logic. Seems the current version of the api does allow null for the resource here